relentless/optimize/optimizer.py

```diff
@@ -0,0 +1,23 @@
+__all__ = ['Optimizer','SteepestDescent']
+
+import abc
+
+from .objective import ObjectiveFunction
+
+class Optimizer(abc.ABC):
+    def __init__(self, obj):
+        self.objective = obj
+
+    @abc.abstractmethod
+    def run(self):
+        pass
+
+class SteepestDescent(Optimizer):
+    def __init__(self, obj, alpha):
+        super().__init__(obj)
+        self.alpha = alpha
+
+    def run(self):
+        res = self.objective.run()
+        for x in res.gradient:
```

Comment:
I think we want to get the design variables first, and iterate over those. It should be the case that the design variables are the same as what's in the result, but I guess this is not necessarily guaranteed. With the new `gradient` method I suggested:
```
vars = self.objective.design_variables()
for x in vars:
    grad = res.gradient(x)
```